client/src/app/plan.service.ts

```diff
@@ -85,12 +91,12 @@ export class PlanService {
           }
         });
 
-        return yearArray;
+        return JSON.parse(JSON.stringify({ years: yearArray }));
       }
       return [];
     };
 
-    return this.userService.getUserData().pipe(map(mapCallback));
```

Comment:
Made a global variable of the user data to be used across the component - initialized in the constructor